specification/src/main/asciidoc/webprofile/jpa-cdi/CDI-JPA.adoc

```diff
@@ -0,0 +1,120 @@
+:cdi-spec: https://jakarta.ee/specifications/cdi/4.1
+:jpa-spec: https://jakarta.ee/specifications/persistence/3.1/jakarta-persistence-spec-3.1
+
+=== Jakarta Persistence & Jakarta Context and Dependency Injection (CDI) Integration
+This section specifies the requirements for Jakarta EE environments that integrate both {jpa-spec}[Persistence] and {cdi-spec}[CDI].
+
+==== Obtaining an Entity Manager using CDI
+
+A container environment might feature built-in integration of
+Jakarta Persistence with the CDI bean manager, allowing injection
+of a container-managed entity manager using the annotation
+_jakarta.inject.Inject_. In the Jakarta EE environment, the
+container must feature built-integration of Jakarta Persistence
+with the CDI bean manager.
+
+For each persistence unit, a container which features such
+built-in integration must make available a bean with:
+
+- bean type _EntityManager_,
+- the qualifiers specified by _qualifier_ XML elements in
+_persistence.xml_, or _jakarta.enterprise.inject.Default_,
+if no qualifiers are explicitly specified,
+- the scope specified by the _scope_ XML element in
+_persistence.xml_, or _jakarta.transaction.TransactionScoped_
+if no scope is explicitly specified,
+- no interceptor bindings,
+- a bean implementation which satisfies the requirements of
+this specification for a container-managed entity manager.
+
+==== Obtaining an Entity Manager Factory using CDI
+
+A container environment might feature built-in integration of
+Jakarta Persistence with the CDI bean manager, allowing
+injection of a container-managed entity manager factory using
+the annotation _jakarta.inject.Inject_. In the Jakarta EE
+environment, the container must feature built-integration of
+Jakarta Persistence with the CDI bean manager.
+
+For each persistence unit, a container which features such
+built-in integration must make available a bean with:
+
+- bean type _EntityManagerFactory_,
+- the qualifiers specified by _qualifier_ XML elements in
+_persistence.xml_, or _jakarta.enterprise.inject.Default_,
+if no qualifiers are explicitly specified,
+- scope _jakarta.enterprise.context.ApplicationScoped_,
+- bean name given by the name of the persistence unit,
+- no interceptor bindings,
+- a bean implementation which satisfies the requirements of
+this specification for a container-managed entity manager
+factory.
+
+Furthermore, the container must make available five beans with:
+
+- bean types _CriteriaBuilder_, _PersistenceUnitUtil_, _Cache_,
+_SchemaManager_, and _Metamodel_, respectively,
```

Comment:
should these be made available for `@Resource` injection and/or DD override (=reachable from JNDI) and as such also some description added to chapter 5?

Reply:
@lukasj anything like that would require mechanisms external to CDI and would significantly complicate this whole thing.

Reply:
(And CDI itself already provides ways to override beans.)

Reply:
If this cannot be made consistent with the rest of resources available through the platform, I would not add this requirement for now

Reply:
> If this cannot be made consistent with the rest of resources available through the platform, I would not add this requirement for now

Or we could document in the Platform Specification what (Jakarta EE 11) portable applications can expect in the Platform Specification and what this new feature is not integrated with (e.g. Resource and/or DD override, I think extended persistence contexts but not completely sure about that).  

Having said that, I think we do need to prototype this requirement in EE implementations so we can see what EE features do not work (for EE implementations) that should work.

Based on this pr conversation, I understand more of what is expected but I think we need to see this feature actually being used so we can all see what happens when very large applications:
- use this feature with active JTA transaction + existing application code also using the same persistence contexts.
- use this feature with  with no active JTA transaction + existing application code.  

Also, the https://jakarta.ee/xml/ns/persistence/persistence_3_2.xsd already has the support for this feature and if we don't define how the Platform uses the new fields, we might end up with a lot different extensions from  the EE vendors, non of which are completely consistent.

Reply:
> should these be made available for @Resource injection and/or DD override (=reachable from JNDI)

I think it's time to let go of @Resource and JNDI. So from my point of view that would be 

-1 -> Do NOT make these available for @Resource injection and/or DD override (=reachable from JNDI)

Reply:
I agree. I don't see the value of `@Resource` or JNDI here. In my mind, that's legacy stuff that we've already moved away from a long time ago.

Reply:
> > should these be made available for @resource injection and/or DD override (=reachable from JNDI)
> 
> I think it's time to let go of @resource and JNDI. So from my point of view that would be
> 
> -1 -> Do NOT make these available for @resource injection and/or DD override (=reachable from JNDI)

then I have to ask - why this functionality has not been deprecated yet? Shouldn't that be the very first step a platform should have taken in order to re-build itself on top of CDI?

Reply:
>  why this functionality has not been deprecated yet? 

There's so many questions there. I think we should have done it in 2009 indeed, but by then the EJB group was still big and powerful, and I think we didn't want to aggitate them too much.

We also shouldn't have just stopped with @Transactional in Java EE 7, but did much more in that release.

It's a sad story alltogether how we coud reinvented ourselves many years ago, but somehow didn't, even though the path was clearly ahead of us.

Java EE/Jakarta Security was the first, in Java EE 8, to add all kinds of resources/services and complete base it on CDI. Completely and utterly ignoring @Resource/JNDI. People though it could not be done. Somehow everything would always need JNDI. But hey, Jakarta Security did it. And now Quarkus and MP also provide tons of services, and it's really true that JNDI is not needed.

Hopefully for EE 12 we can finally deprecate our old friend JNDI and co.

Reply:
We have an issue that was delayed for EE10 on the deprecation of resource injection and JNDI.
https://github.com/jakartaee/platform/issues/355